hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/rapid/RPBSPMC0Converter.java

```diff
@@ -0,0 +1,189 @@
+
+	private String messageDateTime = LocalDateTime.now().format(dateTimeFormatter);
+
+	public RPBSPMC0Converter() {
+		super();
+	}
+
+	@Handler
+	public String convertResponse(String responseMessage, Exchange exchange) {
+		String methodName = LoggingUtil.getMethodName();
+		StringBuffer v2Response = new StringBuffer();
+		RPBSPMC0 rpbspmc0 = new RPBSPMC0(responseMessage);
+		HL7Message mshdefaults = getHL7Message(exchange);
+		R32 r32 = new R32();
+
+		try {
+			populateMSH(r32.getMSH(), mshdefaults);
+			populateMSA(r32.getMSA(), rpbspmc0.getRpbsHeader().getStatusText());
+		} catch (HL7Exception e) {
+			logger.error("{} - TransactionId: {}, Exception while converting R32 data {}", methodName,
+					exchange.getExchangeId(), e.getMessage());
+		}
+		v2Response.append(r32.toString());
+		v2Response.append(
+				"ERR|^^^" + rpbspmc0.getRpbsHeader().getStatusCode() + "&" + rpbspmc0.getRpbsHeader().getStatusText());
+
+		rpbspmc0.getRpbsmc0Data().getBeneficiaries().forEach(b -> {
+			// Output must be a hl7v2 structure so map contract period to a
+			// hl7v2 message
+			try {
+				R32Beneficiary beneficiary = new R32Beneficiary();
+				populateZIA(beneficiary.getZIA(), "", b.getLastName(), b.getFirstName(), "", "", "", "");
+				populatePID(beneficiary.getPID(), b.getPhn());
+				// Remove beneficiary MSH segment
+				StringBuffer formattedResponse = new StringBuffer(beneficiary.toString()).replace(0, 8, "");
+				v2Response.append(formattedResponse.toString());
+
+			} catch (HL7Exception e) {
+				logger.error("{} - TransactionId: {}, Exception while converting beneficiary data {}", methodName,
+						exchange.getExchangeId(), e.getMessage());
+			}
+			b.getContractPeriods().forEach(cp -> {
+				try {
+					R32ContractPeriods contractPeriod = new R32ContractPeriods();
+					populatePID(contractPeriod.getPID(), cp.getPhn());
+					populateNK1(contractPeriod.getNK1(), cp.getRelationship());
+					if (StringUtils.isNotBlank(cp.getReasonCode())) {
+						populateZIH(contractPeriod.getZIH(), cp.getReasonCode());
+					}
+
+					populateIN1(contractPeriod.getIN1(), convertDate(cp.getEffectiveDate()),
+							convertDate(cp.getCancelDate()), cp.getGroupNumber(), "", "");
+					// Remove beneficiary contract period MSH segment
+					StringBuffer formattedResponse = new StringBuffer(contractPeriod.toString()).replace(0, 8, "");
+					v2Response.append(formattedResponse.toString());			
+
+				} catch (HL7Exception e) {
+
+					logger.error("{} - TransactionId: {}, Exception while converting contract period {}", methodName,
+							exchange.getExchangeId(), e.getMessage());
+
+				}
+
+			});
+		});
+
+		logger.debug(v2Response.toString());
+		return v2Response.toString();
+	}
+
+	private HL7Message getHL7Message(Exchange exchange) {
+		// MSH|^~\&|HNWeb|E13DD6BB-69197950C48|RAIGT-CNT-PRDS|BC00001013|20210820125|anu-test|R32|20220920115331|D
+		HL7Message mshdefaults = new HL7Message();
```

Comment:
var naming not camel cased.
Note, these are not default values but rather values from incoming message.